fsw/cfe-core/src/sb/cfe_sb_api.c

```diff
@@ -160,13 +160,27 @@ int32  CFE_SB_CreatePipe(CFE_SB_PipeId_t *PipeIdPtr, uint16  Depth, const char *
 
         /* if OS_QueueCreate() failed because the pipe name passed in was already in use... */
         /* let's make sure we don't alter the user's pipe ID data */
-        if (Status == CFE_OS_ERR_NAME_TAKEN){
-            *PipeIdPtr = OriginalPipeIdParamValue;
-        }
-
-        CFE_EVS_SendEventWithAppID(CFE_SB_CR_PIPE_ERR_EID,CFE_EVS_EventType_ERROR,CFE_SB.AppId,
+        switch(Status) {
+            case CFE_OS_ERR_NAME_TAKEN:
+                CFE_EVS_SendEventWithAppID(CFE_SB_CR_PIPE_ERR_EID,CFE_EVS_EventType_ERROR,CFE_SB.AppId,
+                    "CreatePipeErr:OS_QueueCreate failed, name taken (app=%s, name=%s)",
+                    CFE_SB_GetAppTskName(TskId,FullName), PipeName);
+
+                *PipeIdPtr = OriginalPipeIdParamValue;
+
+                break;
+            case OS_ERR_NO_FREE_IDS:
+                CFE_EVS_SendEventWithAppID(CFE_SB_CR_PIPE_ERR_EID,CFE_EVS_EventType_ERROR,CFE_SB.AppId,
+                    "CreatePipeErr:OS_QueueCreate failed, no free id's (app=%s)",
+                    CFE_SB_GetAppTskName(TskId,FullName));
+
+                break;
+            default:
+                CFE_EVS_SendEventWithAppID(CFE_SB_CR_PIPE_ERR_EID,CFE_EVS_EventType_ERROR,CFE_SB.AppId,
```

Comment:
This should either both use the OSAL-defined code (OS_ERR_NAME_TAKEN, OS_ERR_NO_FREE_IDS) or the CFE-defined equivalent (CFE_OS_ERR_NAME_TAKEN, CFE_OS_ERR_NO_FREE_IDS) but not one of each.

My preference is to use the actual OSAL return codes, the CFE wrapper ones don't add much value.